### DISCUSSION TITLE
iOS v2ray client/lib/package/framework

### DISCUSSION THREAD

**Discussion by P0SHTET** [v2fly/v2ray-core#3219] [Q&A]
I am developing my VPN client for iOS/Android on Flutter using v2ray. I have already implemented the Android part and there are problems with iOS. There are no packages or libraries for Flutter and iOS to implement VPN connection via v2ray. 

I tried to run a binary file, but the iOS system forbids this. 

I also tried to create C libraries and integrate them into my application via FFI, but it didn't work out either

To create.xcframework using gomobile also fails.

Can anyone help me here?

我正在使用v2ray在Flutter上为iOS/Android开发我的VPN客户端。 我已经实现了Android部分，并且iOS存在问题。 Flutter和iOS没有通过v2ray实现VPN连接的包或库。 

我试图运行一个二进制文件，但iOS系统禁止这样做。 

我也试图创建C库并通过FFI将它们集成到我的应用程序中，但它也没有成功

来创造。使用gomobile的xcframework也失败。

有人能帮我吗？

tg: https://t.me/poshtet
email: pmponomarenko1003@gmail.com

**Comment by the-thirteenth-fox** [comment]
https://github.com/XIIIFOX/flutter_vless - complete free and working nice both iOS and android
